src/tools/paint_surface.cc

```diff
@@ -155,7 +155,7 @@ struct paint_surface_tool : tool
                 c[i[1]] = k;
 
                 auto block = ship->get_block(cur);
-                if (mode == replace_mode::match && block->surfs[start_index] != select_type) {
+                if (block->type != block_frame || (mode == replace_mode::match && block->surfs[start_index] != select_type)) {
```

Comment:
`if (!block || ... all of this` for the missing chunk case.

Reply:
oh, right